OpenQA/Isotovideo/Utils.pm

```diff
@@ -64,18 +93,27 @@ sub clone_git ($local_path, $clone_url, $clone_depth, $branch, $dir, $dir_variab
         bmwqemu::fctinfo "Checking out git refspec/branch '$branch'";
         $branch_args = " --branch $branch";
     }
+
     my $clone_cmd = 'env GIT_SSH_COMMAND="ssh -oBatchMode=yes" git clone';
-    my @out = qx{$clone_cmd --depth=$clone_depth $branch_args $clone_url 2>&1};
     my $handle_output = sub ($return_value, @out) {
         bmwqemu::diag "@out" if @out;
         die "Unable to clone Git repository '$dir' specified via $dir_variable (see log for details)" unless $return_value == 0;
         return 1;
     };
-    return $handle_output->($?, @out) unless ($branch && grep /warning: Could not find remote branch/, @out);
+
+    my $cache_dir = _handle_caching($clone_url, $clone_depth, $branch, $clone_cmd, $handle_output);
+    $clone_url = $cache_dir if $cache_dir;    # clone from the cache dir instead
+
+    # attempt to clone with `--branch`
+    my $depth_args = $cache_dir ? '' : "--depth='$clone_depth'";    # cannot use `--depth` with $cache_dir
```

Comment:
why can we not use `--depth`?

Reply:
I don't remember the exact error message anymore but Git complained about it when using it to clone from a local directory. The clone from the real repo will still use the `--depth` parameter as before so this shouldn't cause more traffic.